__tests__/app.test.js

```diff
@@ -294,4 +294,27 @@ describe('GET /api/users', () => {
             })
         })
     })
+})
+
+describe('GET /api/articles (topic query)', () => {
+    test('200: responds with an array of articles that are related to the queried topic', () => {
+        return request(app).get('/api/articles?topic=cats').expect(200).then(({ body }) => {
+            expect(body.articles).toHaveLength(1);
+        })
+    })
+    test('200: responds with an array of all articles if query is omitted', () => {
+        return request(app).get('/api/articles').expect(200).then(({ body }) => {
+            expect(body.articles).toHaveLength(13);
+        })
+    })
+    test('404: responds with an error message when there are no articles found with the associated topic', () => {
+        return request(app).get('/api/articles?topic=paper').expect(404).then(({ body }) => {
+            expect(body.msg).toBe('No articles found with that topic');
+        })
+    })
+    test('400: responds with an error message when passed topic is invalid', () => {
+        return request(app).get('/api/articles?topic=notvalid').expect(400).then(({ body }) => {
```

Comment:
This test makes everything mega difficult so well done for including it and making it pass!